spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/reactor/TraceReactorAutoConfiguration.java

```diff
@@ -94,10 +81,6 @@ static HookRegisteringBeanDefinitionRegistryPostProcessor traceHookRegisteringBe
 	}
 }
 
-interface LastOperatorWrapper {
```

Comment:
Why have you removed that?

Reply:
Just trying to figure out the reason why tests fail on CI environment as I can't reproduce this on local env.
JFY: LastOperatorWrapper  code seems to be not covered by tests. When I removed this code no new tests failed. Don't you know what this code (LastOperatorWrapper) for?

Reply:
The concept is such that if you have a non web application we want to apply a different last operator wrapper. It seems that we're missing a test for that. We should add it instead of removing this functionality

Reply:
Not going to merge this change just experimenting with build on CI. Can you point me to an issue or case that will describe why we need this additional hook on last operator for not web application?

Reply:
For non web application we assume that if you're calling reactor's flux, you're doing it manually. In that case we want to close the span when the flux is finished.

In a web application we want a filter to close the flux or any other higher level component. We don't want multiple components try to star, close spans and report them to Zipkin.

Reply:
Don't know all the corner cases but it seems that with new Mono/Flux  `@NewSpan` and `@ConinueSpan` annotations processing span get started after subscription and closed on flux termination

Reply:
So maybe you fixed it! In fact there's a test that should fail if something goes wrong https://github.com/spring-cloud/spring-cloud-sleuth/blob/master/spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/reactor/SpanSubscriberTests.java . Do you have a WebFlux based app which you can click to see that your changes don't break anything?  If not you can use this https://github.com/spring-cloud-samples/sleuth-documentation-apps

Reply:
Thanks! Replaced `Tracing.current().currentTraceContext().get().spanId()` 
with `tracer.currentSpan().context().spanId()` as in `SpanSubscriberTest` and get tests fixed !!!